qa/tasks/deepsea.py

```diff
@@ -363,18 +467,61 @@ def _introspect_roles(self, deepsea_ctx):
         for var in report_vars:
             self.log.info("{} == {}".format(var, deepsea_ctx[var]))
 
+    def _master_python_version(self, py_version):
+        """
+        Determine if a given python version is installed on the Salt Master
+        node.
+        """
+        python_binary = 'python{}'.format(py_version)
+        installed = True
+        try:
+            self.master_remote.run(args=[
+                'type',
+                python_binary,
+                run.Raw('>'),
+                '/dev/null',
+                run.Raw('2>&1'),
+                ])
+        except CommandFailedError:
+            installed = False
+        if installed:
+            self.master_remote.run(args=[
+                python_binary,
+                '--version'
+                ])
+        else:
+            self.log.info(
+                '{} not installed on master node'.format(python_binary)
+                )
+        return installed
+
+    def _maybe_apply_alternative_defaults(self):
+        global_yml = '/srv/pillar/ceph/stack/global.yml'
+        if self.alternative_defaults:
+            data = data = "\n".join([
+                yaml.dump(x, default_flow_style=False) for x in self.alternative_defaults
+                ])
+            if data:
+                # remote.sh does not return partial output when the command times out,
+                # so we hope the following command will not time out
+                self.master_remote.sh("sudo tee -=a {}".format(global_yml), input=data)
+        dump_file_that_might_not_exist(self.master_remote, global_yml)
```

Comment:
I lord, why do we need this function? Isn't following code self documented?
```
self.master_remote.sh('sudo cat ' + global_yaml + ' || true')
```

Reply:
Honestly, I don't see why repeating that code is more desirable than having a dedicated function.

Reply:
@smithfarm I have no strong opinion regarding this, but having weird long sentence for a function which does trivial self-documented one-line instruction looks overkill for me. If you wish to be precise as much as possible, to reflect the right sense of this method it can be named:
```
dump_file_that_might_not_exist_on_a_remote_host_using_sudo_permissions
```
However, maybe one of the following names `sudo_cat`, `quiet_cat` or `remote_quiet_cat` is more convenient? :-)